lib/activerecord-bitemporal/bitemporal.rb

```diff
@@ -329,6 +329,10 @@ def _update_row(attribute_names, attempted_action = 'update')
           self.transaction_from = after_instance.transaction_from
           self.transaction_to = after_instance.transaction_to
 
+          if attempted_action == 'touch'
+            @_touch_attr_names += Set.new(['valid_from', 'valid_to', 'transaction_from', 'transaction_to'])
```

Comment:
Append to an instance variable used by `ActiveRecord::AttributeMethods::Dirty#_touch_row`.
https://github.com/rails/rails/blob/v7.0.6/activerecord/lib/active_record/attribute_methods/dirty.rb#L187

This is probably not a good approach, but it's the only way to write out to the `saved_changes` at once with minimal changes.